java/src/main/java/cucumber/runtime/java/AbstractJavaSnippet.java

```diff
@@ -24,6 +25,9 @@ public final String arguments(Map<String, Type> arguments) {
     private String getArgType(Type argType) {
         if (argType instanceof Class) {
             Class cType = (Class) argType;
+            if (cType.equals(DataTable.class)) {
```

Comment:
So you always want the fqn?

Reply:
Yes. But you don't have to share that opinion. 

I reckon it will be nice for the same reason the data table import is nice. It does make snippets more verbose.

Have you tried to see which gives you the best experience?

Reply:
Personally, I don't like the fully qualified names; I think it's ugly. And tbh I have no problem just importing PendingException and DataTable. 
But if it makes it easier for people to use Cucumber, I don't mind having fqns. Although I'd prefer to have them only when really needed.

Reply:
Some context: We train people in Cucumber using [Cyber-Dojo](http://www.cyber-dojo.org/) - an online coding environment for TDD and kata practices.

It doesn't add import statements automatically like an IDE does. So I thought it would be easier for people if class names were FQN.

Under normal circumstances I think FQN would drive me nuts.

So I think we should make it configurable. I'm wary of the inflation of command line options (annotation options), but that's probably the right way to do it.

When we integrate [config](https://github.com/cucumber/cucumber/tree/master/config) people can change this setting in a file.

Reply:
Fair enough!

Reply:
@aslakhellesoy Thanks, that does clarify the need for me ;) Making it configurable will be a lot more work though.

Reply:
Yes, and it *shouldn't* be a lot of work. Let's postpone this until we're using `config`, which takes care of the hard work. With that in place it should be easy.

Reply:
I don't think the the problem will be in the config.

It will be quite cumbersome to get the config information to where it is needed. The snippet generator is currently created in the backend.

It might be useful to start looking at some DI solution to create the backends.